### PR TITLE
fix: wrong xo config resolved

### PR DESCRIPTION
**Pull request by JounQin**
### Initial checklist

* [x] I read the support docs 
* [x] I read the contributing guide 
* [x] I agree to follow the code of conduct 
* [x] I searched issues and couldn’t find anything (or linked relevant results below) 
* [x] If applicable, I’ve added docs and tests

### Description of changes

This PR fixes a lot of CI errors which is caused by https://github.com/xojs/xo/issues/741

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -41,6 +41,7 @@
     "create-gfm-fixtures": "^1.0.0",
     "hastscript": "^7.0.0",
     "jsdom": "^21.0.0",
+    "patch-package": "^8.0.0",
```
Comment (JounQin) [inline package.json]: temporary workaround

**Comment by JounQin** [comment]
https://github.com/rehypejs/rehype-github/blob/7d7e851b751413c1ec95e3fbdf1c48e4b71155e5/packages/break/package.json#L37

@wooorm OK, why `xo: false` is set? That's the root cause of the CI test errors. Please help to confirm the desired behavior so that I can update this PR accordingly.

**Comment by wooorm** [comment]
Because previously `xo` used `xo: false` for nested packages, to continue looking for a parent package.json.
Omitting `xo: false`, previously, resulted in the defaults being applied.
Which is exactly what xo is now doing when `xo: false` is used.

**Comment by wooorm** [comment]
Note: updating this project is complex, you don’t have to do the work, I can do it too!

**Comment by JounQin** [comment]
> Because previously `xo` used `xo: false` for nested packages, to continue looking for a parent package.json. Omitting `xo: false`, previously, resulted in the defaults being applied. Which is exactly what xo is now doing when `xo: false` is used.

So this seems a new bug which I described at https://github.com/xojs/xo/issues/741#issuecomment-1820127993, I'll change the fix then.

**Comment by JounQin** [comment]
> Note: updating this project is complex, you don’t have to do the work, I can do it too!

No worry, I just found bugs and want to fix them, no matter it's in `xo` or `rehype-github`, just interesting.

**Comment by JounQin** [comment]
image

The only left thing is `test-coverage`, I think I'll leave it to you and you can fix the `test-coverage` based on this PR. @wooorm

**Comment by JounQin** [comment]
It's also fine if you want to close this PR.

**Comment by wooorm** [comment]
Fixed! Thanks <3

**Pull request closed without merging**